### ISSUE TITLE
Don’t lose statues being composed after switching apps

### ISSUE THREAD

**Issue by fluffy-critter** [DagAgren/toot#420]
Sometimes I start composing a toot and need to switch to another app to copy-paste something or to check a fact or something, and when I come back the app will have reset and lost the toot in progress. Doing something to save the composer state when switching away would be really nice.

Relatedly, being able to save a toot as a draft for later (like in the official twitter app) in case there’s a momentary networking or instance problem.

**Comment by DagAgren** [comment]
This is a bit UI-design-heavy so not a huge priority at the moment, but it would probably be useful.

**Comment by fluffy-critter** [comment]
The don't-lose-a-toot-I'm-composing thing is the important thing to me; the saved drafts are more of an afterthought that would be nice to have. I think not losing a toot when switching apps wouldn't need any change in UI, just in behavior.

Like, the particular situation that led to opening this issue: I was composing a toot that needed some fiddly Unicode characters that weren't available from any of my configured keyboards, and switching to Safari to copy-paste them from Wikipedia led to my in-progress toot being totally lost.

**Comment by mediaformat** [comment]
Yes I've lost many a good Toot! after answering the phone, the phone becoming locked or a number of other reasons. An auto save state, or auto save drafts would be very useful!

**Comment by DagAgren** [comment]
Just switching apps or answering calls should not close the app and lose the toot. That is a bit odd. Are your phones very memory restrained or something?

**Comment by fluffy-critter** [comment]
You can't name assumptions about what situations might cause proves death in iOS. I don't use toot anymore (for mental health reasons) but when I did it was on a current iPhone without any particularly contained memory situation, but sometimes just switching to a browser and doing a web search can thrash things a lot. Consider how much crap a modern website loads.

**Comment by ellieayla** [comment]
Bitten by the lack of this again today. :(
